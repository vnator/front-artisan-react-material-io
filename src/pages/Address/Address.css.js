import React from 'react';

import {
  Table as MUITable,
  TableCell,
  TableContainer,
  styled,
  Button as MUIButton,
  TextField,
  Typography,
} from '@material-ui/core';

const TabContainer = styled(TableContainer)({
  display: 'flex',
  justifyContent: 'center',
  margin: '40 0',
});

const Table = styled(MUITable)({
  maxWidth: 'calc(100% - 80px)',
});

const Footer = styled(({ children, ...props }) => (
  <div {...props}>{children}</div>
))({
  width: '100%',
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
  flexDirection: 'column',
  paddingBottom: 40,
});

const FooterRow = styled(({ children, ...props }) => (
  <div {...props}>{children}</div>
))(props => ({
  display: 'flex',
  justifyContent: 'space-between',
  alignItems: 'center',
  width: 'calc(100% - 80px)',
  border: `1px solid ${props.theme.palette.primary.main}`,
  borderTop: 0,
}));

const Input = styled(TextField)({
  flexGrow: 1,
});

const Button = styled(MUIButton)({
  height: 53.3,
  width: 160,
  borderRadius: 0,
});

const Cell = styled(TableCell)(props => ({
  width: 40,
  background: props.theme.palette.grey['200'],
}));

const Head = styled(TableCell)(props => ({
  backgroundColor: props.theme.palette.primary.main,
  color: props.theme.palette.common.white,
}));

const Title = styled(Typography)({
  marginBottom: 40,
});

export {
  Cell,
  Button,
  Input,
  FooterRow,
  Footer,
  Table,
  TabContainer,
  Head,
  Title,
};
