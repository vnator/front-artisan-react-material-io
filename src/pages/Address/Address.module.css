.Address {
  --border: 1px solid var(--white-full);

  flex-grow: 1;
  display: flex;
  text-align: center;
  align-items: center;
  justify-content: center;
  background-color: var(--first-dark);
}

.addressList {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.addressItem {
  height: 32px;
  border: var(--border);
  border-bottom-width: 0;
  display: flex;
  justify-content: space-between;
  min-width: 80vw;
}

.addressList .addressItem:last-child {
  border-bottom-width: 1px;
}

.key {
  width: 8vw;
  min-width: 120px;
  display: flex;
  align-items: center;
  justify-content: center;
  border: 0;
  background: var(--white-full);
  color: var(--black-full);
  cursor: pointer;
  transition: var(--transition)
}

.key:hover {
  opacity: 0.8;
}

.value {
  color: var(--white-full);
  flex-grow: 1;
  text-align: left;
  padding: var(--space-base);
  box-sizing: border-box;
  display: block;
  align-items: center;
  border: 0;
  background: none;
  border-radius: 0;
}

._index {
  max-width: 80px;
  background: var(--black-full);
  text-align: center;
}

.input {
  background: var(--grey-dark);
} 
