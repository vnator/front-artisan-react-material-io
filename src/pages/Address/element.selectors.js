const el = Object.freeze({
  btnSelectStreet: 'test-address-select_street',
  selectedStreet: 'test-address-selected_street',
  inputUpdateStreet: 'test-address-input_update_street',
  btnUpdateStreet: 'test-address-btn_update_street',
  inputUpdateCity: 'test-address-input_update_city',
  btnUpdateCity: 'test-address-btn_update_city',
});

export { el };
