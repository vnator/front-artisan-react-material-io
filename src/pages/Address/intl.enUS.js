const address_enUS = {
  street: 'street',
  city: 'city',
  submit: 'submit',
  placeholder: {
    street: 'update row street name',
    city: 'Set new city for all address',
  },
};

export { address_enUS };
