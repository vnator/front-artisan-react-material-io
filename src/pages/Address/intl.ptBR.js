const address_ptBR = {
  street: 'street',
  city: 'city',
  submit: 'submit',
  placeholder: {
    street: 'update row street name',
    city: 'Set new city for all address',
  },
};

export { address_ptBR };
