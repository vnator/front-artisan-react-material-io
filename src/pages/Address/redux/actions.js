import { createAction } from '@reduxjs/toolkit';
import { address } from './types';

const _setCity = ({ value, index }) => {
  return {
    payload: {
      index: [index, 'city'],
      value,
    },
  };
};

const _setStreet = ({ value, index }) => {
  return {
    payload: {
      index: [index, 'street'],
      value,
    },
  };
};

const _setNumber = ({ value, index }) => ({
  payload: {
    index: [index, 'number'],
    value,
  },
});

const setStreet = createAction(address.SET_STREET, _setStreet);
const setCity = createAction(address.SET_CITY, _setCity);
const setNumber = createAction(address.SET_NUMBER, _setNumber);

export { setStreet, setCity, setNumber };
