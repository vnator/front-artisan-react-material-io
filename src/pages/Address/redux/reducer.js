import { createReducer, createAction } from '@reduxjs/toolkit';
import { assocPath } from 'ramda';
import { address } from './types';

const setStreet = createAction(address.SET_STREET);
const setCity = createAction(address.SET_CITY);
const setNumber = createAction(address.SET_NUMBER);

const _initialState = Object.freeze({
  first: {
    city: 'Curitiba',
    street: 'Rua dos Boticarios',
    number: '12',
  },
  second: {
    city: 'Ponta Grossa',
    street: 'Marcondes',
    number: '54',
  },
  third: {
    city: 'Laranjal',
    street: 'Maria Ap. Serodio',
    number: '32',
  },
  fourty: {
    city: 'Curitiba',
    street: 'Barao do Rio Branco',
    number: '1232',
  },
});

const addressReducer = createReducer(_initialState, {
  [setCity]: (state, action) => {
    const { index, value } = action.payload;
    return assocPath(index, value, state);
  },

  [setStreet]: (state, action) => {
    const { index, value } = action.payload;
    return assocPath(index, value, state);
  },

  [setNumber]: (state, action) => {
    const { index, value } = action.payload;
    return assocPath(index, value, state);
  },
});

export { addressReducer, _initialState };
