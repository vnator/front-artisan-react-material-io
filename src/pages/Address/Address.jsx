import React, { Component } from 'react';
import Type from 'prop-types';
import { EditOutlined } from '@material-ui/icons';

import {
  TableBody,
  TableCell,
  TableHead,
  TableRow,
  IconButton,
} from '@material-ui/core';

import {
  Table,
  TabContainer,
  Head,
  Footer,
  FooterRow,
  Cell,
} from '../../components/Table.css';
import { Title } from '../../components/Title.css';
import { intlShape } from '../../utils/intlShape';

import { Button, Input } from './Address.css';
import { el } from './element.selectors.js';

class Address extends Component {
  // selector example usage
  state = {
    selected: '',
    newAddress: '',
    newCity: '',
  };

  updateStreet = () =>
    this.props.setStreet({
      value: this.state.newAddress,
      index: this.state.selected,
    });

  render() {
    const { formatMessage } = this.props.intl;
    const { address } = this.props;

    return (
      <div>
        <Title variant="h4" color="primary">
          {formatMessage({ id: 'address.title' })}
        </Title>
        <TabContainer>
          <Table aria-label="simple table" size="small">
            <TableHead>
              <TableRow>
                <Head size="medium">
                  {formatMessage({ id: 'address.head.id' })}
                </Head>
                <Head size="medium">
                  {formatMessage({ id: 'address.head.city' })}
                </Head>
                <Head size="medium">
                  {formatMessage({ id: 'address.head.street' })}
                </Head>
                <Head size="medium">
                  {formatMessage({ id: 'address.head.number' })}
                </Head>
                <Head size="medium"></Head>
              </TableRow>
            </TableHead>
            <TableBody>
              {Object.entries(address).map(([key, val]) => (
                <TableRow key={key}>
                  <TableCell>{key}</TableCell>
                  <TableCell>{val.city}</TableCell>
                  <TableCell>{val.street}</TableCell>
                  <TableCell>{val.number}</TableCell>
                  <Cell>
                    <IconButton
                      id={el.btnSelectStreet}
                      color="primary"
                      onClick={() => {
                        this.setState(prev => ({
                          ...prev,
                          selected: key,
                          newAddress: val.street,
                        }));
                      }}>
                      <EditOutlined />
                    </IconButton>
                  </Cell>
                </TableRow>
              ))}
            </TableBody>
          </Table>
        </TabContainer>

        <Footer>
          <FooterRow>
            <Input
              id={el.inputUpdateStreet}
              value={this.state.newAddress}
              onChange={e => this.setState({ newAddress: e.target.value })}
              variant="filled"
              label={
                this.state.selected
                  ? Object.keys(this.props.address).find(
                      x => x === this.state.selected,
                    )
                  : formatMessage({ id: 'address.street' })
              }
              placeholder={formatMessage({
                id: 'address.placeholder.street',
              })}
              inputProps={{ 'aria-label': 'search google maps' }}
            />
            <Button
              variant="contained"
              color="primary"
              id={el.btnUpdateStreet}
              disableElevation
              onClick={this.updateStreet}>
              {formatMessage({ id: 'address.submit' })}
            </Button>
          </FooterRow>
          <FooterRow>
            <Input
              variant="filled"
              label={formatMessage({ id: 'address.city' })}
              id={el.inputUpdateCity}
              value={this.state.newCity}
              placeholder={formatMessage({
                id: 'address.placeholder.city',
              })}
              onChange={e => this.setState({ newCity: e.target.value })}
              inputProps={{ 'aria-label': 'search google maps' }}
            />
            <Button
              variant="contained"
              color="primary"
              id={el.btnUpdateCity}
              disableElevation
              onClick={() => this.props.setAddressCity(this.state.newCity)}>
              {formatMessage({ id: 'address.submit' })}
            </Button>
          </FooterRow>
        </Footer>
      </div>
    );
  }
}

Address.propTypes = {
  intl: Type.shape(intlShape).isRequired,
  setStreet: Type.func.isRequired,
  setAddressCity: Type.func.isRequired,
  address: Type.objectOf(
    Type.shape({
      city: Type.string,
      street: Type.string,
      number: Type.string,
    }),
  ).isRequired,
};

export { Address };
