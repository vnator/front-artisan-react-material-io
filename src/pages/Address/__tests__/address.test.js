import React from 'react';
import Enzyme, { mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { spy } from 'sinon';

import { Address } from '../Address';
import { _initialState } from '../redux/reducer';
import { el } from '../element.selectors';
import { intl } from '../../../utils/enzymeHelper';

Enzyme.configure({ adapter: new Adapter() });

describe('Address', () => {
  const mockProps = {
    address: _initialState,
    setStreet: spy(),
    setAddressCity: spy(),
    intl,
  };

  beforeEach(() => {
    mockProps.setAddressCity.resetHistory();
    mockProps.setStreet.resetHistory();
  });

  it('Snapshot', () => {
    const address = mount(<Address {...mockProps} />);

    expect(address).toMatchSnapshot();
  });

  it('onClick select address', () => {
    const address = shallow(<Address {...mockProps} />);
    const button = address.find(`.${el.btnSelectStreet}`).first();

    button.simulate('click');
    expect(address.state().selected).toBe(button.props().children);
  });

  it('onChange newStreet', () => {
    const newAddress = 'carambolas';
    const address = shallow(<Address {...mockProps} />);
    const input = address.find(`.${el.inputUpdateStreet}`);

    input.simulate('change', { target: { value: newAddress } });
    expect(address.state().newAddress).toBe(newAddress);
  });

  it('onClick updateStreet', () => {
    const address = shallow(<Address {...mockProps} />);
    const button = address.find(`.${el.btnUpdateStreet}`);

    button.simulate('click');
    expect(mockProps.setStreet.calledOnce).toBeTruthy();
  });

  it('onChange newCity', () => {
    const newCity = 'Marcelandia';
    const address = shallow(<Address {...mockProps} />);
    const input = address.find(`.${el.inputUpdateCity}`);

    input.simulate('change', { target: { value: newCity } });
    expect(address.state().newCity).toBe(newCity);
  });

  it('onClick setAddressCity', () => {
    const address = shallow(<Address {...mockProps} />);
    const button = address.find(`.${el.btnUpdateCity}`);

    button.simulate('click');
    expect(mockProps.setAddressCity.calledOnce).toBeTruthy();
  });
});
