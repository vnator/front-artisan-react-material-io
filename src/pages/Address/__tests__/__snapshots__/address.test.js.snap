// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Address Snapshot 1`] = `ReactWrapper {}`;
